README.md

```diff
@@ -29,6 +29,52 @@ To run the tutorials or the test suite locally, run the following to install the
 pip install -e .[dev]
 ```
 
+## Instanciation and Authentication
+
+There are several ways to provide a correct authentication using the SDK.
+
+```python
+from sdk import SDK
+
+group_id="your_group_id" # Replace this value by your group_id on the PASQAL platform.
+some_token="token"
+username="your_username" # Replace this value by your username or email on the PASQAL platform.
+password="your_password" # Replace this value by your password on the PASQAL platform.
+# Ideally, do not write this password in a script but provide in through the command-line or as a secret environment variable.
+
+""" Method 1: Username + Password
+    If you know your credentials, you can pass them to the SDK instance on creation.
+"""
+sdk = SDK(username=username, password=password, group_id=group_id)
+
+""" Method 2: Username only
+    If you only want to insert your username, but want a solution to have your password being secret
+    you can run the SDK without password. A prompt will then ask for your password
+"""
+sdk = SDK(username=username, group_id=group_id)
+> Please, enter your password:
+
+""" Method 3: Use a token.
+    If you already know your token, you can directly pass it as an argument.
+"""
+sdk = SDK(token_provider=token, group_id=group_id)
+```
+
+If you want to redefine the APIs used by the SDK, please, do the following.
+
+```python
+from sdk import SDK, Endpoints
+
+class MyNewEndpoints(Endpoints):
```

Comment:
Yes, but I think it should be recommended to use this feature this way because we don't want to override every endpoints.
Plus, we now have an Endpoint we may not want to change (especially in local mode, since it does not exist) : Auth0

Reply:
But we can just modify specific values in the Endpoints, right?

Reply:
Same point as above for the Token, I think since the feature exists and is an entrypoint in the application, we should document it. Yes, for me it is trivial to change the `Endpoints` in the library if I need to make a change using the `pip install -e`, but this feature exists specifically to override the Endpoints, so, I think we should either delete the feature, or document it properly (i.e : give a how to use example)

Reply:
Yes, to override the core endpoint and keep the other as defaults you can simply do `endpoints = Endpoints(core=<my-endpoint>)`

Reply:
So, do you prefer I document it like this ?

```python
new_endpoints = Endpoints(core = "http://new_url:new_port")
```

Reply:
Yeah, that's exactly what I meant too. Sorry for being unspecific!